anomalib/models/draem/augmenter.py

```diff
@@ -0,0 +1,150 @@
+"""Augmenter module to generates out-of-distribution samples for the DRAEM implementation."""
+
+# Original Code
+# Copyright (c) 2022 VitjanZ
+# https://github.com/VitjanZ/DRAEM.
+# SPDX-License-Identifier: MIT
+#
+# Modified
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+import glob
+import random
+from typing import Optional, Tuple
+
+import cv2
+import imgaug.augmenters as iaa
+import numpy as np
+import torch
+from torch import Tensor
+
+from anomalib.models.draem.perlin import rand_perlin_2d_np
+
+
+class Augmenter:
+    """Class that generates noisy augmentations of input images.
+
+    Args:
+        anomaly_source_path (Optional[str]): Path to a folder of images that will be used as source of the anomalous
+        noise. If not specified, random noise will be used instead.
+    """
+
+    def __init__(self, anomaly_source_path: Optional[str] = None):
+
+        if anomaly_source_path is not None:
+            self.anomaly_source_paths = sorted(glob.glob(anomaly_source_path + "/**/*.jpg", recursive=True))
+        else:
+            self.anomaly_source_paths = []
+
+        self.augmenters = [
+            iaa.GammaContrast((0.5, 2.0), per_channel=True),
+            iaa.MultiplyAndAddToBrightness(mul=(0.8, 1.2), add=(-30, 30)),
+            iaa.pillike.EnhanceSharpness(),
+            iaa.AddToHueAndSaturation((-50, 50), per_channel=True),
+            iaa.Solarize(0.5, threshold=(32, 128)),
+            iaa.Posterize(),
+            iaa.Invert(),
+            iaa.pillike.Autocontrast(),
+            iaa.pillike.Equalize(),
+            iaa.Affine(rotate=(-45, 45)),
+        ]
+        self.rot = iaa.Sequential([iaa.Affine(rotate=(-90, 90))])
```

Comment:
Is it possible to use `albumentations` library instead of `imgaug`? They are similar libraries, would be good to use one of them for consistency

Reply:
I agree. I tried rewriting this with albumentations, but some of these transforms are not available in the albumentations package. For reproducibility I decided to keep imgaug for now. I don't expect it to lead to significant performance drop if we omit or replace some of these transforms, but it would be good to verify this first.